src/vaadin-radio-group-mixin.ts

```diff
@@ -0,0 +1,330 @@
+import { FocusVisibleMixin, FocusVisibleInterface } from '@vaadin/focus-visible-mixin/focus-visible-mixin.js';
+import { FocusVisibleClass } from '@vaadin/focus-visible-mixin/focus-visible-class.js';
+import {
+  KeyboardDirectionMixin,
+  KeyboardDirectionInterface
+} from '@vaadin/keyboard-direction-mixin/keyboard-direction-mixin.js';
+import { KeyboardDirectionClass } from '@vaadin/keyboard-direction-mixin/keyboard-direction-class.js';
+import { RovingTabIndexMixin } from '@vaadin/roving-tabindex-mixin/roving-tabindex-mixin.js';
+import { SlottedItemsMixin, SlottedItemsInterface } from '@vaadin/slotted-items-mixin/slotted-items-mixin.js';
+import { RadioButtonMixinClass } from './vaadin-radio-button-mixin-class';
+import { RadioButton } from './vaadin-radio-button-mixin';
+
+export interface RadioGroupInterface {
+  value: string | null | undefined;
+
+  label: string | null | undefined;
+
+  invalid: boolean | null | undefined;
+
+  required: boolean | null | undefined;
+
+  readonly: boolean | null | undefined;
+
+  errorMessage: string | null | undefined;
+
+  orientation: string | null | undefined;
+
+  checkValidity(): boolean;
+
+  validate(): boolean;
+}
+
+type RadioGroup = RadioGroupInterface &
+  SlottedItemsInterface &
+  KeyboardDirectionInterface &
+  FocusVisibleInterface &
+  DisabledStateInterface &
+  FocusVisibleClass &
+  KeyboardDirectionClass;
+
+const filterRadioButtons = (nodes: Node[]) => {
+  return nodes.filter(node => {
+    return (
+      node.nodeType === Node.ELEMENT_NODE && (node.constructor as typeof RadioButtonMixinClass).hasRadioButtonMixin
+    );
+  }) as HTMLElement[];
+};
+
+let uniqueId = 0;
+
+export const RadioGroupMixin = <T extends Constructor<LitElement & KeyboardDirectionClass>>(
+  base: T
+): T & Constructor<RadioGroup> => {
+  class RadioGroup extends FocusVisibleMixin(
+    DisabledStateMixin(RovingTabIndexMixin(KeyboardDirectionMixin(SlottedItemsMixin(base))))
+  ) {
+    /**
+     * String used for the label element.
+     */
+    @property({ type: String }) label: string | null | undefined;
+
+    /**
+     * Value of the radio group.
+     */
+    @property({ type: String }) value: string | null | undefined;
+
+    /**
+     * When true, the radio group value is invalid.
+     */
+    @property({ type: Boolean, reflect: true }) invalid: boolean | null | undefined = false;
+
+    /**
+     * Radio group is aligned vertically by default. Set to "horizontal" to change this.
+     */
+    @property({ type: String, reflect: true }) orientation: string | null | undefined;
+
+    /**
+     * When true, the radio group must have a value set by the user.
+     */
+    @property({ type: Boolean }) required: boolean | null | undefined;
+
+    /**
+     * When true, the user cannot modify the value of the radio group.
+     */
+    @property({ type: Boolean }) readonly: boolean | null | undefined;
```

Comment:
Nit: Should `required` and `readonly` be reflected as well?

Reply:
Done.